hadoop-resource/src/test/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetailsTest.java

```diff
@@ -1,171 +1,173 @@
-    }
-}
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.resource.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import uk.gov.gchq.palisade.resource.LeafResource;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class HadoopResourceDetailsTest {
+
+    @BeforeEach
+    public void setup() {
+        HadoopResourceDetails.addTypeSupport("type", "TYPE");
+    }
+
+    @Test
+    void acceptsSchemelessUri() throws URISyntaxException {
+        // Given
+        URI uri = new URI("/home/hadoop/resources/type_file.format");
+        HadoopResourceDetails expected = new HadoopResourceDetails(uri, "TYPE", "format");
+
+        // When
+        HadoopResourceDetails details = HadoopResourceDetails.getResourceDetailsFromFileName(uri);
+
+        // Then
+        assertThat(details).isEqualTo(expected);
+    }
+
+    @Test
+    void acceptsAbsolutePath() throws URISyntaxException {
+        // Given
+        URI absolute = new URI("file:/home/hadoop/resources/type_file.format");
+        HadoopResourceDetails expected = new HadoopResourceDetails(absolute, "TYPE", "format");
+
+        // When
+        HadoopResourceDetails details = HadoopResourceDetails.getResourceDetailsFromFileName(absolute);
+
+        // Then
+        assertThat(details).isEqualTo(expected);
+    }
+
+    @Test
+    void acceptsRelativePath() throws URISyntaxException {
+        // Given
+        URI relative = new URI("file:./type_file.format");
+        HadoopResourceDetails expected = new HadoopResourceDetails(relative, "TYPE", "format");
+
+        // When
+        HadoopResourceDetails details = HadoopResourceDetails.getResourceDetailsFromFileName(relative);
+
+        // Then
+        assertThat(details).isEqualTo(expected);
+    }
+
+    @Test
+    void noTypeThrowsException() {
+        // Given
+        URI invalidType = new File(".").toURI().resolve("file.format");
+
+        // When
+        assertThrows(IllegalArgumentException.class,
+                () -> HadoopResourceDetails.getResourceDetailsFromFileName(invalidType));
+
+        // Then throw
+    }
```

Comment:
Update tests to follow the correct naming convention
Add an assertion on the message returned from the thrown exception

```suggestion
    @Test
    void testNoTypeThrowsException() {
        // Given
        URI invalidType = new File(".").toURI().resolve("file.format");

        // When the resource is retrieved with an invalid format
        Exception illegalAccessException = assertThrows(IllegalArgumentException.class,
                () -> HadoopResourceDetails.getResourceDetailsFromFileName(invalidType), "Test should throw an exception");

        // Then check the assertion message
        assertThat(illegalAccessException.getMessage()).contains("Filename doesn't comply with TYPE_FILENAME.FORMAT");
    }
```

Reply:
Updated